src/permissionSelectors.js

```diff
@@ -40,6 +42,105 @@ export function roleIsAtLeast(thisRole: RoleT, thresholdRole: RoleT): boolean {
   return thisRole <= thresholdRole; // Roles with more privilege have lower numbers.
 }
 
+/**
+ * Whether the user has passed the realm's waiting period to be a full member.
+ *
+ * See:
+ *   https://zulip.com/api/roles-and-permissions#determining-if-a-user-is-a-full-member
+ *
+ * To determine if a user is a full member, callers must also check that the
+ * user's role is at least Role.Member.
+ *
+ * Note: If used with useSelector, the value will be out-of-date, though
+ * realistically only by seconds or minutes at most; see implementation.
+ */
+export function getHasUserPassedWaitingPeriod(state: PerAccountState, userId: UserId): boolean {
+  const { waitingPeriodThreshold } = getRealm(state);
+  const { date_joined } = getUserForId(state, userId);
+
+  return (
+    differenceInDays(
+      // When used with useSelector, the result will be based on the time at
+      // which the most recent Redux action was dispatched. This would break
+      // if we made this a caching selector; don't do that.
+      // TODO(?): To upper-bound how long ago that can be, we could dispatch
+      //   actions at a regular short-ish interval. If those actions
+      //   contained a current-date value, we could even store that value in
+      //   Redux and consume it here, letting this be a caching selector if
+      //   we wanted it to be.
```

Comment:
I see. Yeah, this is fine. These waiting periods are days long, and once you've crossed it you've crossed it -- so it'll be very uncommon that the staleness matters.

Web has a bug where Recent Topics routinely tells me a conversation was "Just now" or "3 minutes ago" when in reality it was like 20 minutes ago -- I've just returned to the Zulip tab after doing other things, nothing had happened since then to force it to update, and it never bothered to update the relative times. That bug should be fixed. This potential bug would be a much smaller issue :smile: